aten/src/ATen/native/TensorTransformations.h

```diff
@@ -33,4 +33,20 @@ static inline void flip_check_errors(int64_t total_dims, int64_t flip_dims_size,
     ", but unique flip dims size=", flip_dims_v.size());
 }
 
+static inline Tensor roll_common(const Tensor& self, IntList shifts, IntList dims) {
+  if (dims.size() == 0 && shifts.size() == 1) {
+    auto flattened = self.contiguous().view(self.numel());
+    return roll(flattened, shifts[0], 0).view(self.sizes());
+  }
+  AT_CHECK(
+    shifts.size() == dims.size(),
+    "shifts and dimensions must align. shifts: ", shifts.size(), ", dims:", dims.size()
+  );
+  AT_CHECK(dims.size() > 1, "this code should only be reached for handling multi-dim rolling" );
+  auto tail_shifts = shifts.slice(1);
+  auto tail_dims = dims.slice(1);
+  auto first_dim_rolled = roll(self, shifts[0], dims[0]);
```

Comment:
is there a better way than rolling one dim at a time?

Reply:
It seems like all dimensions could be rolled at once by a single cuda kernel using the current kernel logic, just by iterating through the dimensions to update the shift amount. We'd need to pass the array of dims/shifts every time, which could be less efficient for the more common case of a single dimension. Maybe the optimal implementation will use two different kernels depending on number of dims being rolled?

I haven't thought about optimizations to the CPU version.

Either way, I'd prefer the functionality working, and possibly iterate on performance later.

Reply:
Two different kernels sounds reasonable in the future.